cmd/tailscale/cli/switch.go

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package cli
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/peterbourgon/ff/v3/ffcli"
+	"tailscale.com/ipn"
+)
+
+var switchCmd = &ffcli.Command{
+	Name:       "switch",
+	ShortHelp:  "Switches to a different Tailscale profile",
+	LongHelp:   `"tailscale switch" switches between logged in profiles`,
+	ShortUsage: "switch [<name>] [--list]",
+	FlagSet: func() *flag.FlagSet {
+		fs := flag.NewFlagSet("switch", flag.ExitOnError)
+		fs.BoolVar(&switchArgs.list, "list", false, "list available profiles")
+		return fs
+	}(),
+	Exec: switchProfile,
+}
+
+var switchArgs struct {
+	list bool
+}
+
+func listProfiles(ctx context.Context) error {
+	curP, all, err := localClient.ProfileStatus(ctx)
+	if err != nil {
+		return err
+	}
+	for _, prof := range all {
+		if prof.ID == curP.ID {
+			fmt.Printf("%s *\n", prof.Name)
+		} else {
+			fmt.Println(prof.Name)
+		}
+	}
+	return nil
+}
+
+func switchProfile(ctx context.Context, args []string) error {
+	if switchArgs.list {
+		return listProfiles(ctx)
+	}
+	if len(args) != 1 {
+		outln("usage: tailscale profile switch NAME")
+		os.Exit(1)
+	}
+	cp, all, err := localClient.ProfileStatus(ctx)
+	if err != nil {
+		printf("Failed to switch to profile: %v\n", err)
+		os.Exit(1)
+	}
+	var profID ipn.ProfileID
+	for _, p := range all {
+		if p.Name == args[0] {
+			profID = p.ID
+			break
+		}
+	}
+	if profID == "" {
+		printf("No profile named %q\n", args[0])
+		os.Exit(1)
+	}
+	if profID == cp.ID {
+		printf("Already on profile %q\n", args[0])
+		os.Exit(0)
+	}
+	if err := localClient.SwitchProfile(ctx, profID); err != nil {
+		printf("Failed to switch to profile: %v\n", err)
+		os.Exit(1)
+	}
+	printf("Switched to profile %q\n", args[0])
```

Comment:
Switched or Switching? Looks like it might still fail based on following for loop.